### PR TITLE
Fix SauceConnect when the proxy fails in various ways

### PR DESCRIPTION
**Pull request by gsnedders**
I still don't like the sleep loop; suggestions welcome!

---
This change is [Reviewable](https://reviewable.io/reviews/w3c/web-platform-tests/9201)

### CODE REVIEW

File: tools/wptrunner/wptrunner/browsers/sauce.py [hunk 1]
```diff
@@ -156,10 +156,19 @@ def __enter__(self, options):
             "web-platform.test",
             "*.web-platform.test"
         ])
-        while not os.path.exists('./sauce_is_ready') and not self.sc_process.poll():
-            time.sleep(5)
 
-        if self.sc_process.returncode is not None and self.sc_process.returncode > 0:
+        tot_wait = 0
+        while not os.path.exists('./sauce_is_ready') and self.sc_process.poll() is None:
+            time.sleep(5)
```
Comment (jgraham) [inline tools/wptrunner/wptrunner/browsers/sauce.py]: Put this in a variable and use it n your counter.

**Review by jgraham — APPROVED** [review]

File: tools/wptrunner/wptrunner/browsers/sauce.py (hunk 1 above)
Comment (gsnedders) [reply tools/wptrunner/wptrunner/browsers/sauce.py]: Does this look better now?

**Comment by w3c-bots** [comment]
# Build BROKEN

Started: 2018-01-25 20:50:37
Finished: 2018-01-25 21:06:15


<h2>Failing Jobs</h2>
<ul>

<li>tools_unittest in py27</li>

<li>tools_unittest in pypy</li>

</ul>




View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/23303)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/333444228)

**Comment by gsnedders** [comment]
23:20 < jgraham> gsnedders: Happy for you to land if you want

So, squash & merge when CI completes.

**Pull request merged**